content/posts/Docker Tags.md

```diff
@@ -0,0 +1,54 @@
+---
+title: "Deciding which Docker tag to use"
+date: 2019-03-30T12:00:00Z
+aliases:
+    - /posts/2018-03-30-docker-tags/
+---
+
+In this post we're going to talk about how versioning works in Docker container images. We’ll also explore some different strategies you will want to consider when using image version tags in your own Dockerfiles.
+
+Docker container names are split into 2 parts: the name of the container image, and the version tag of the container image. For example, the container `python:3` is a container hosted on Docker Hub at [https://hub.docker.com/_/python](https://hub.docker.com/_/python). The name of the container is `python`, and `3` is the version tag. There are several other version tags of `python` also available, such as:
+
+python:3.7,
+python:3.7.3
+python:3-alpine
+python:3.7-alpine
+python:3.7.3-alpine
+python:slim
+etc.
+
+Most of the popular languages use a similar tagging pattern in their official images on Docker Hub. Each of these tags represents different variations of an install of python. Some use a different base operating system images (alpine VS debian VS “slimmed-down” debian), and they also indicate the specific versions of whatever language is installed in the image.
+
+Today the `python:3` container image specifically uses python version 3.8. Whenever 3.9 is released, that tag will get updated and the `python:3` image will then contain version 3.9 instead. In a similar way, the `python:3.7` image would be the latest available security patch of python 3.7. Currently that happens to be 3.7.3, but tomorrow that could be 3.7.4 if a new version is released.
+
+When writing your own applications, you will need to decide which version of the base image(s) you will use in the `FROM` statement(s) of your Dockerfile(s). The options include:
+
+- None (just use :latest, for example, just `FROM python`)
```

Comment:
Clarify this a bit—we have some language in the book about the semantics of `latest` which you could excerpt here.